src/rust/engine/options/src/config.rs

```diff
@@ -346,53 +427,54 @@ impl OptionsSource for Config {
         f64::from_config(self, id)
     }
 
+    fn get_bool_list(&self, id: &OptionId) -> Result<Option<Vec<ListEdit<bool>>>, String> {
+        self.get_list(id, parse_bool_list)
+    }
+
+    fn get_int_list(&self, id: &OptionId) -> Result<Option<Vec<ListEdit<i64>>>, String> {
+        self.get_list(id, parse_int_list)
+    }
+
+    fn get_float_list(&self, id: &OptionId) -> Result<Option<Vec<ListEdit<f64>>>, String> {
+        self.get_list(id, parse_float_list)
+    }
+
     fn get_string_list(&self, id: &OptionId) -> Result<Option<Vec<ListEdit<String>>>, String> {
+        self.get_list(id, parse_string_list)
+    }
+
+    fn get_dict(&self, id: &OptionId) -> Result<Option<DictEdit>, String> {
+        let mut dict_edit: Option<DictEdit> = None;
         if let Some(table) = self.config.get(id.scope()) {
             let option_name = Self::option_name(id);
-            let mut list_edits = vec![];
             if let Some(value) = table.get(&option_name) {
                 match value {
                     Value::Table(sub_table) => {
-                        if sub_table.is_empty()
-                            || !sub_table.keys().collect::<HashSet<_>>().is_subset(
-                                &["add".to_owned(), "remove".to_owned()]
-                                    .iter()
-                                    .collect::<HashSet<_>>(),
-                            )
-                        {
-                            return Err(format!(
-                "Expected {option_name} to contain an 'add' element, a 'remove' element or both but found: {sub_table:?}"
-              ));
-                        }
                         if let Some(add) = sub_table.get("add") {
-                            list_edits.push(ListEdit {
-                                action: ListEditAction::Add,
-                                items: String::extract_list(&format!("{option_name}.add"), add)?,
-                            })
-                        }
-                        if let Some(remove) = sub_table.get("remove") {
-                            list_edits.push(ListEdit {
-                                action: ListEditAction::Remove,
-                                items: String::extract_list(
-                                    &format!("{option_name}.remove"),
-                                    remove,
-                                )?,
-                            })
+                            if sub_table.len() == 1 && add.is_table() {
```

Comment:
What happens if these aren't true? It looks like the value is silently dropped?

Reply:
> What happens if these aren't true? It looks like the value is silently dropped?

Ah yes, good catch. We should probably never have supported a subtable named "add" the way we do for lists, because unlike in the list case this is also a valid dict value. But we do in Python so we must here as well. 

Anyway, fixed.